### PR TITLE
Mask password in Openstack VIM driver logs

### LINKED ISSUE #128: Mask password in Openstack VIM driver logs
Mask password in Openstack VIM driver logs

### PR DESCRIPTION
**Pull request by Akshata-Desai-ibm**
Fixes #128

### CODE REVIEW

**Review by lokanalla — CHANGES_REQUESTED** [review]
Could you try to add a unit test to verify the password is really masked!

**Comment by Akshata-Desai-ibm** [comment]
> Could you try to add a unit test to verify the password is really masked!

I have added some unit test cases to verify the password is masked

File: osvimdriver/openstack/heat/template.py [hunk 1]
```diff
@@ -23,6 +25,20 @@ def __filter_from_dictionary(self, parameters, properties_dict):
             if k in properties_dict:
                 used_properties[k] = properties_dict[k]
         return used_properties
+    
+    def filter_password_from_dictionary(self, heat_template_str):
+        heat_tpl = yaml.safe_load(heat_template_str)
+        if 'resources' in heat_tpl:
+            resources = heat_tpl['resources'] 
+            user_data = resources['apache_server']['properties']['user_data']
```
Comment (lokanalla) [inline osvimdriver/openstack/heat/template.py:33]: This code is very specific to apache_server properties. What if there are any other types of servers with password property in it. Please generalise the code to be applicable for all data in the the heat template.

**Review by lokanalla — CHANGES_REQUESTED** [review]

File: osvimdriver/openstack/heat/template.py (hunk 1 above)
Comment (Akshata-Desai-ibm) [reply osvimdriver/openstack/heat/template.py]: Generalised the code to be applicable for all data in the the heat template.

File: tests/unit/openstack/heat/test_template.py [hunk 2]
```diff
@@ -3,6 +3,24 @@
 from ignition.utils.propvaluemap import PropValueMap
 
 class TestHeatInputUtil(unittest.TestCase):
+    def setUp(self):
```
Comment (lokanalla) [inline tests/unit/openstack/heat/test_template.py:6]: Can't this variables be declared within the test case as these are not being used in any other tests?

**Review by lokanalla — CHANGES_REQUESTED** [review]

File: tests/unit/openstack/heat/test_template.py (hunk 2 above)
Comment (Akshata-Desai-ibm) [reply tests/unit/openstack/heat/test_template.py:6]: I have used this variable in following function

```
def test_filter_password_from_dictionary(self):
        result = self.obj.filter_password_from_dictionary(self.template)
        self.assertEqual(result, self.filtered_template) 
```
And also used in other test cases

**Review by lokanalla — APPROVED** [review]

File: tests/unit/openstack/heat/test_template.py [hunk 3]
```diff
@@ -3,6 +3,24 @@
...
+        self.template = '''
```
[the remark this answers is not in the document]
Comment (lokanalla) [reply tests/unit/openstack/heat/test_template.py]: Akshata has moved the code appropriately.

**Pull request merged**